app/src/main/java/com/droidhats/campuscompass/models/Location.kt

```diff
@@ -0,0 +1,109 @@
+package com.droidhats.campuscompass.models
+
+import android.util.Log
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Polygon
+import com.google.android.gms.maps.model.PolygonOptions
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/*
+* Model for location classes
+* Data relating to locations should be stored in this file
+* */
+
+abstract class Location(coordinate: LatLng) {
+    private var coordinate: LatLng = coordinate
+}
+
+// Model for Campus class
+class Campus(coordinate: LatLng, name: String, jsonObject: JSONObject) : Location(coordinate) {
+    private var buildingsList: MutableList<Building> = mutableListOf()
+    private val name: String = name
+    private val jsonObject: JSONObject = jsonObject
+    fun getName(): String = name
+    fun getBuildings(): MutableList<Building> = buildingsList
```

Comment:
I mean, why we're returning this instead of something immutable. Are we going to be needing to change this list later? It might be safer to keep it immutable (through a regular List) so that other classes don't mutate this list.

Reply:
I had it as a regular List initially but in Kotlin, once the List is created you can't add/remove elements to it (there is no add() method)
so basically during parsing the json at createBuildings(), I wasn't able to add buildings one by one as I traversed them -> for the same reason that List was immutable 
the only workaround is to use MutableList unfortunately (I checked the API)
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-list/index.html

Reply:
actually now that I think about it, there is another workaround ;) 
fixed it